a3p-integration/proposals/z:acceptance/test-lib/psm-lib.js

```diff
@@ -0,0 +1,492 @@
+import '@endo/init';
+import { boardSlottingMarshaller, makeFromBoard } from './rpc.js';
+import {
+  addUser,
+  agd,
+  agops,
+  agoric,
+  executeOffer,
+  getUser,
+  agopsLocation,
+  executeCommand,
+  CHAINID,
+  VALIDATORADDR,
+  GOV1ADDR,
+  mkTemp,
+} from '@agoric/synthetic-chain';
+import { AmountMath } from '@agoric/ertp';
+import { getBalances } from './utils.js';
+import { PSM_PAIR, USDC_DENOM } from '../psm.test.js';
+import fsp from 'node:fs/promises';
+
+const fromBoard = makeFromBoard();
+const marshaller = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+
+const VOTING_WAIT_MS = 65 * 1000;
+
+/**
+ *  Import from synthetic-chain once it is updated
+ *
+ * @param {string} addr
+ * @param {string} wanted
+ * @param {string} [from]
+ */
+export const bankSend = (addr, wanted, from = VALIDATORADDR) => {
+  const chain = ['--chain-id', CHAINID];
+  const fromArg = ['--from', from];
+  const testKeyring = ['--keyring-backend', 'test'];
+  const noise = [...fromArg, ...chain, ...testKeyring, '--yes'];
+
+  return agd.tx('bank', 'send', from, addr, wanted, ...noise);
+};
+
+/**
+ *
+ * @param {{
+ *   address: string
+ *   instanceName: string
+ *   newParams: Params
+ *   deadline: number
+ *   offerId?: string
+ * }} QuestionDetails
+ */
+export const buildProposePSMParamChangeOffer = async ({
+  address,
+  instanceName,
+  newParams,
+  deadline,
+  offerId = Date.now().toString(),
+}) => {
+  const charterAcceptOfferId = await agops.ec(
```

Comment:
_suggestion; feel free to ignore_

`agops` uses ambient authority. synthetic-chain doesn't have a handy alternative, but in dapp-orchestration-basics, one technique I recently used is for the caller to do:

```
  const execFileP = promisify(childProcess.execFile);
  const npx = (file: string, args: string[]) =>
    execFileP('npx', ['--no-install', file, ...args]);
```
https://github.com/Agoric/dapp-orchestration-basics/blob/main/e2e-testing/scripts/deploy-cli.ts

and then here we would do
```suggestion
  const charterAcceptOfferId = await npx('agops' , ['ec,
```

Reply:
ignored for now